### DISCUSSION TITLE
[multiSchema previewFeature] The database schema is not empty error

### DISCUSSION THREAD

**Discussion by reza-ebrahimi** [prisma/prisma#21005] [Q&A] [answered]
### Bug description

Here is my schema, after having a fesh schema and table in postgres `prisma migrate dev` throws following error:
`The database schema is not empty`

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["multiSchema"]
}

datasource db {
  provider  = "postgresql"
  url       = env("POSTGRES_PRISMA_URL")
  directUrl = env("POSTGRES_URL_NON_POOLING")
  schemas   = ["app"]
}

model NewTable {
  id BigInt @id

  @@schema("app")
}
```

Reproduce:

```bash
$ manually created an schema and a table in pgAdmin
$ npx prisma db pull
$ npx prisma migrate dev
....
Error: P3005

The database schema is not empty. Read more about how to baseline an existing production database: https://pris.ly/d/migrate-baseline
```

### How to reproduce

```bash
$ manually created an schema and a table in pgAdmin
$ npx prisma db pull
$ npx prisma migrate dev
....
Error: P3005

The database schema is not empty. Read more about how to baseline an existing production database: https://pris.ly/d/migrate-baseline
```

### Prisma information

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["multiSchema"]
}

datasource db {
  provider  = "postgresql"
  url       = env("POSTGRES_PRISMA_URL")
  directUrl = env("POSTGRES_URL_NON_POOLING")
  schemas   = ["app"]
}

model NewTable {
  id BigInt @id

  @@schema("app")
}
```

### Environment & setup

- OS: Ubuntu
- Database: PostgreSQL
- Node.js version: v20.3.0

### Prisma Version

```bash
$ npx prisma version
prisma                  : 5.2.0
@prisma/client          : 5.2.0
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine 2804dc98259d2ea960602aca6b8e7fdc03c1758f (at ../../node_modules/.pnpm/@prisma+engines@5.2.0/node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Schema Engine           : schema-engine-cli 2804dc98259d2ea960602aca6b8e7fdc03c1758f (at ../../node_modules/.pnpm/@prisma+engines@5.2.0/node_modules/@prisma/engines/schema-engine-debian-openssl-1.1.x)
Schema Wasm             : @prisma/prisma-schema-wasm 5.2.0-25.2804dc98259d2ea960602aca6b8e7fdc03c1758f
Default Engines Hash    : 2804dc98259d2ea960602aca6b8e7fdc03c1758f
Studio                  : 0.494.0
Preview Features        : multiSchema
```

**Comment by aqrln** [comment]
Thanks for reaching out and opening the issue. You are getting this error because, as you said, you created the db schema manually. You need to do what the error message says and create a baseline migration to be able to use Prisma Migrate with the existing non-empty database: https://pris.ly/d/migrate-baseline

**Reply by aqrln** [reply]
Alternatively, since you said it was just a fresh schema, is there any reason you created it manually? If not, you can just with an empty database and let Prisma create the database schema according to your Prisma schema, as it's the normal workflow for new projects.

**Reply by reza-ebrahimi** [reply]
No, I created manually because of the errors I get before almost for every command I executed.

**Reply by reza-ebrahimi** [reply]
Why I need to follow baseline steps for a fresh schema?

**Reply by reza-ebrahimi** [reply]
I also followed the prisma steps to create from scratch and it gives the same error, please change this back to issue.

```
$ npx prisma init --datasource-provider postgresql
$ # adding multiSchema feature
$ npx prisma migrate dev --name init
Environment variables loaded from .env
Prisma schema loaded from schema.prisma
Datasource "db": PostgreSQL database "postgres", schemas "app" at "localhost:5432"

Error: P3005

The database schema is not empty. Read more about how to baseline an existing production database: https://pris.ly/d/migrate-baseline
```

**Reply by reza-ebrahimi** [reply]
@aqrln

**Reply by ludralph** [reply]
Hi @reza-ebrahimi 👋 

Can you take a look at this [comment](https://github.com/prisma/prisma/issues/19234#issuecomment-1560722661) and this [workaround](https://github.com/prisma/prisma/issues/19234#issuecomment-1574095096)

**Reply by ludralph** [reply]
You can also try this [workaround](https://github.com/prisma/prisma/issues/19234#issuecomment-1574816962) as well.

**Reply by reza-ebrahimi** [reply]
@ludralph Thanks, but there is no shadow database url in my schema. The problem is with multiSchema feature.

**Reply by reza-ebrahimi** [reply]
Just to add, everything works fine when I remove the `multiSchema` feature from schema.

```prisma
generator client {
  provider        = "prisma-client-js"
}

datasource db {
  provider  = "postgresql"
  url       = env("POSTGRES_PRISMA_URL")
  directUrl = env("POSTGRES_URL_NON_POOLING")
}

model NewTable {
  id BigInt @id
}
```

**Reply by ludralph** [reply]
@reza-ebrahimi Are you using Vercel Postgres by any means or just a local pgAdmin database?

**Reply by reza-ebrahimi** [reply]
No, its just local.

**Reply by aqrln** [reply]
> Just to add, everything works fine when I remove the multiSchema feature from schema.

This difference in behaviour doesn't sound right, I'll convert it back to an issue.

**Reply by nikolasburk** [reply, marked as answer]
I'll mark this Discussion as answered since the conversation is continued here: https://github.com/prisma/prisma/issues/21007
